slm_porosity/heat_transfer.f90

```diff
@@ -780,6 +798,9 @@ SUBROUTINE user_pre_process
     REAL(pr), DIMENSION(dim) :: max_grad_temp, thickness, resolution
     LOGICAL :: interface_mask(nwlt)
 
+    IF ((.NOT.half_domain .AND. dim.EQ.3) .AND. &
+      (initial_laser_position(2).LE.xyzlimits(1,2) .OR. initial_laser_position(2).GE.xyzlimits(2,2))) &
+      STOP 'Please check your laser position and domain boundary'
```

Comment:
Because there is no symmetry in 2D domain.
There are `half_domain` flag implementation and checking of this flag usability. I think they strictly connected.
OK